go/types/struct.go

```diff
@@ -195,13 +210,45 @@ func EscapeStructField(input string) string {
 		buf.WriteString(hs)
 		return buf.String()
 	}
+	output := EscapeFields(input, encode)
+	verifyFieldName(output)
```

Comment:
We should have thorough unit tests for functions, then callers of those functions can assume they are correct without having to double-check the work.